core/src/com/etheller/warsmash/viewer5/handlers/w3x/simulation/behaviors/build/CBehaviorHumanRepair.java

```diff
@@ -85,25 +85,31 @@ protected CBehavior update(CSimulation simulation, boolean withinFacingWindow) {
                         targetUnit.setConstructionPowerBuild(true);
                     }
                     if(costs_gold > player.getGold() || costs_lumber > player.getLumber()) {
+
                         return unit.pollNextOrderBehavior(simulation);
                     }
                     targetUnit.setConstructionProgress(targetUnit.getConstructionProgress() +constructionProgressGain);
                 }
 
                 newLifeValue = Math.min(targetUnit.getLife() + healthGain, targetUnit.getMaximumLife());
                 if(costs_gold > player.getGold() || costs_lumber > player.getLumber()){
+
                     return unit.pollNextOrderBehavior(simulation);
                 }
-                boolean done = (newLifeValue >= targetWidget.getMaxLife());
+                boolean done = (newLifeValue >= ((CUnit) targetWidget).getUnitType().getMaxLife());
```

Comment:
As a general rule, we should probably avoid `getUnitType().getMaxLife()` except during unit creation and when upgrading a unit into a different type, simply because of the possibility of heroes with life added by strength, or buildings with the Improved Masonry upgrade, etc.